src/ReactDOM.re

```diff
@@ -477,6 +477,33 @@ external unmountComponentAtNode: Dom.element => unit =
 [@mel.module "react-dom"]
 external flushSync: (unit => unit) => unit = "flushSync";
 
+module Experimental = {
+  [@deriving jsProperties]
+  type preOptions = {
+    [@mel.as "as"] [@mel.optional]
+    _as: option([ | `script]),
```

Comment:
I see other variants of "as": 

audio, document, embed, fetch, font, image, object, script, style, track, video, worker.

Reply:
@davesnx Yes, if you reference with MDN. for React is only `script`. See this https://react.dev/reference/react-dom/preloadModule#parameters

Reply:
That's only on the preloadModule and preinitModule.

preinit supports style and scripts: https://react.dev/reference/react-dom/preinit (which also has `precedence`)

while preload has the entire list I pasted above: https://react.dev/reference/react-dom/preload

Reply:
@davesnx I have been add new commit for `preinit` options.


57d426b